hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJetInstance.java

```diff
@@ -170,7 +170,6 @@ public JetCacheManager getCacheManager() {
 
     @Override
     public void shutdown() {
-        observables.values().forEach(Observable::destroy);
```

Comment:
this does some other logic like removing listeners, this will be done automatically when client shutdowns I assume?

Reply:
No, listeners aren't killed on `JetInstance.shutdown()` now either, but indeed, they should be. Then the behaviour would be more consistent with what `getObservables()` is supposed to achieve too. I'll revisit this whole thing again and take care of the listeners.

Reply:
I think Listeners are killed automatically from server side when client disconnects

Reply:
You were right, listeners (even though they are local to the client) get killed on `JetInstance` shutdown (regardless if it's a client or a member), because they get a client/instance inactive exception and cancel themselves because of it. 

Have added a test for this. 

Initial change is enough, we don't need to do more.